### DISCUSSION TITLE
Softmax per graph for edges output

### DISCUSSION THREAD

**Discussion by antoni4040** [pyg-team/pytorch_geometric#4736] [Q&A]
I have something like this:
```
src, dst = edge_index
score = torch.cat((actorX[src], actorX[dst], actorX1[src], actorX1[dst], 
            actorX2[src], actorX2[dst], actorX3[src], actorX3[dst]), 1)
```
And then I pass those pair-wise features through some MLP layers. The output has shape (num_edges, 4).
Then, I want to pass this through a softmax per graph in the batch.
`score = softmax(score, data.batch[src])`
The problem is that this is doing softmax per column (1,2,3,4), but I need it to do it for all of them together. If I flatten it, the shape doesn't match with the data.batch. Any ideas how to overcome this?

**Comment by rusty1s** [comment]
Can you clarify what you want to do exactly? In general, `softmax` is only applied on a single dimension. As an alternative, you can have two MLPs: One that maps to `(num_edges, 4)` as before, and one that maps to `(num_edges, 1)`. You can then use the second output to re-weight the first one:
```python
out = self.mlp1(...)
score = self.ml2(...)
score = softmax(score, data.batch[src])
return out * score
```
